### PR TITLE
Design process manager API

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @rasendubi
See #6

### CODE REVIEW

File: procman/include/procman/procman.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+typedef uint32_t pid_t;
+
+// Forward declarations
+typedef struct intrusive_list_t intrusive_list_t;
+typedef struct thread_t thread_t;
+typedef struct threadman_t threadman_t;
+typedef struct scheduler_t scheduler_t;
+
+enum {
+    PROC_NAME_MAX = 128 /**< Max length for process name. */
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: What's point of using enum? (just curious)

File: procman/include/procman/procman.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+typedef struct process {
+    // TODO: Add list node member as soons as intrusive lists implementation arrives.
+    pid_t pid;
+    struct process *parent; /**< Pointer to parent proces, can be NULL. */
+    intrusive_list_t *children; /**< List of child processes. */
+    char name[PROC_NAME_MAX+1];
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: Better make `PROC_NAME_MAX` equal to `127`, so the structure is aligned.

File: procman/include/procman/procman.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#ifndef PROCMAN_H_
+#define PROCMAN_H_
+
+#include <stdint.h>
+
+// TODO: is 32-bits sufficient? Should we use signed values?
+typedef uint32_t pid_t;
+
+// Forward declarations
+typedef struct intrusive_list_t intrusive_list_t;
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: use just `list_t`

File: procman/include/procman/procman.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
+#ifndef PROCMAN_H_
+#define PROCMAN_H_
+
+#include <stdint.h>
+
+// TODO: is 32-bits sufficient? Should we use signed values?
+typedef uint32_t pid_t;
+
+// Forward declarations
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: Stupid comment for C toddlers.

File: procman/include/procman/procman.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    // TODO: Extend this structure with list of file descriptors and VM page table.
+} process_t;
+
+typedef struct procman {
+    intrusive_list_t *processes;
+    threadman_t *threadman;
+    scheduler_t *scheduler;
+} procman_t;
+
+int procman_init(procman_t **out);
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: Does it allocate memory? If no, use a single pointer. If yes—it shouldn't.

Comment (rasendubi) [inline procman/include/procman/procman.h]: `out` is bad parameter name.

File: procman/include/procman/procman.h [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * TODO: add error_t and error codes.
+ */
+int process_create(procman_t *procman, process_t **out, process_t *parent, const char *name);
+
+void process_delete(procman_t *procman, process_t *process);
+
+int process_add_thread(procman_t *procman, process_t *process, thread_t *thread);
+
+process_t *process_get_py_pid(procman_t *procman, pid_t pid);
```
Comment (rasendubi) [inline procman/include/procman/procman.h]: s/py/by

File: procman/include/procman/scheduler.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+typedef struct scheduler {
+    // TODO
+} scheduler_t;
+
+int scheduler_init(scheduler_t **out /* TODO: some strategy enum ?? */);
+
+void scheduler_destroy(scheduler_t *schduler);
+
+scheduler_params_t *sched_params_create(/* TODO */);
```
Comment (rasendubi) [inline procman/include/procman/scheduler.h]: `scheduler`?

File: procman/include/procman/procman.h (hunk 1 above)
Comment (RostakaGmfun) [inline procman/include/procman/procman.h]: Just a fancy way to define constants

**Comment by RostakaGmfun** [comment]
@rasendubi Thanks for the review!

This PR can be merged only after the following issues will be resolved:
* #9 Intrusive list.
* #10 error_t type and error codes.

File: CMakeLists.txt [hunk 8]
```diff
@@ -4,13 +4,13 @@ project(n00b0s C ASM-ATT)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 add_subdirectory(platform)
-add_subdirectory(core)
+add_subdirectory(procman)
 
 # CMake does not allow executables with no source files
 file(WRITE ${CMAKE_BINARY_DIR}/blank.c "")
```
Comment (forGGe) [inline CMakeLists.txt]: Sure it does. See interface libraries for details.

**Pull request merged**